### PR TITLE
create metrics after plugin

### PR DESCRIPTION
**Pull request by DonutEspresso**

### CODE REVIEW

**Comment by DonutEspresso** [comment]
CI will fail until restify/node-restify#1251 lands

File: README.md [hunk 1]
```diff
@@ -105,6 +102,33 @@ The module includes the following response plugins:
...
+  * `[options.printLog]` {Boolean} when true, prints audit logs. default true.
+* `metrics(callback)` - a metrics plugin which will invoke callback with the
+  the following parameters:
+  * `err` {Object} an error if the request had an error
+  * `metrics` {Object} - metrics about the request
+  * `metrics.statusCode` {Number} status code of the response
+  * `metrics.method` {String} http request verb
+  * `metrics.latency` {Number} request latency
+  * `metrics.url` {String} req.path() value
+  * `metrics.connectionState` {String} can be either 'closed' or 'aborted'. If
```
Comment (yunong) [inline README.md]: s/closed/close

File: lib/plugins/metrics.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+            statusCode: res.statusCode,
+            // REST verb
+            method: req.method,
+            // overall request latency
+            latency: Date.now() - req._time,
+            // the cleaned up url path
+            // e.g., /foo?a=1 => /foo
+            path: req.path(),
+            // connection state can currently only have the following values:
+            // 'close' | 'aborted'
```
Comment (yunong) [inline lib/plugins/metrics.js]: or null

File: test/metrics.test.js [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+var helper = require('./lib/helper');
+var plugins = require('../lib');
+
+// local globals
+var SERVER;
+var CLIENT;
+var PORT;
+
+
+describe('query parser', function () {
```
Comment (yunong) [inline test/metrics.test.js]: query parser?

**Review by yunong — CHANGES_REQUESTED** [review]
A few documentation bugs. Overall looks good.

**Comment by DonutEspresso** [comment]
Comments addressed, one last time?

File: lib/plugins/metrics.js (hunk 2 above)
Comment (yunong) [reply lib/plugins/metrics.js]: Still haven't updated this in the docs?

Comment (yunong) [reply lib/plugins/metrics.js]: e.g. close | aborted | null

File: lib/plugins/metrics.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+    assert.object(opts, 'opts');
+    assert.object(opts.server, 'opts.server');
+    assert.func(callback, 'callback');
+
+    return function metrics(req, res, route, err) {
+
+        var data = {
+            // response status code
+            statusCode: res.statusCode,
```
Comment (yunong) [inline lib/plugins/metrics.js:36]: probably best to add a comment here about this being null too, and how it gets here.

File: lib/plugins/metrics.js (hunk 2 above)
Comment (DonutEspresso) [reply lib/plugins/metrics.js]: Oh shoot, updated the README didn't update this. Default value is undefined

**Comment by DonutEspresso** [comment]
Third time's the charm?

**Review by yunong — APPROVED** [review]

**Pull request merged**